metal/loadbalancers/emlb/emlb.go

```diff
@@ -10,24 +10,62 @@ import (
 )
 
 type LB struct {
-	client *lbaas.APIClient
+	client     *lbaas.APIClient
+	controller *controller
 }
 
 func NewLB(k8sclient kubernetes.Interface, config string) *LB {
+	metalAPIKey := "TODO"
 	lb := &LB{}
-	emlbConfig := lbaas.NewConfiguration()
-	lb.client = lbaas.NewAPIClient(emlbConfig)
+	lb.controller = NewController(metalAPIKey)
 	return lb
 }
 
 func (l *LB) AddService(ctx context.Context, svcNamespace, svcName, ip string, nodes []loadbalancers.Node) error {
+	// 1. Gather the properties we need: Metal API key, port number(s), cluster name(?), target IP(s?)
+	additionalProperties := map[string]string{}
+
+	// 2. Create the infrastructure (what do we need to return here?  lb name and/or ID? anything else?)
+	_, err := l.controller.createLoadBalancer(ctx, additionalProperties)
+
+	if err != nil {
+		return err
+	}
+
+	// 3. Add the annotations
 	return nil
 }
 
 func (l *LB) RemoveService(ctx context.Context, svcNamespace, svcName, ip string) error {
+	// 1. Gather the properties we need: Metal API key, port number(s), cluster name(?), target IP(s?)
+	loadBalancerId := "TODO"
+	additionalProperties := map[string]string{}
+
+	// 2. Delete the infrastructure (do we need to return anything here?)
```

Comment:
possibly delete the IP.

If a user brings an existing IP reservation (or range), the IP wouldn't be deleted.

Reply:
https://github.com/kubernetes-sigs/cloud-provider-equinix-metal/issues/44

Reply:
seems we don't need to worry about the IP here, it is negotiated in https://github.com/kubernetes-sigs/cloud-provider-equinix-metal/blob/main/metal/eip_controlplane_reconciliation.go

Reply:
We can't bring an existing IP reservation today.